src/req_wrap-inl.h

```diff
@@ -121,6 +121,7 @@ struct MakeLibuvRequestCallback<ReqT, void(*)(ReqT*, Args...)> {
 
   static void Wrapper(ReqT* req, Args... args) {
     ReqWrap<ReqT>* req_wrap = ReqWrap<ReqT>::from_req(req);
+    req_wrap->MakeWeak();
     req_wrap->env()->DecreaseWaitingRequestCounter();
```

Comment:
I think this is what’s causing the test failures – based on the stack traces, moving these two calls to the end of the function here should fix this?

Reply:
The MakeWeak() call in particular makes the object prone to getting collected prematurely, I think?

Reply:
Moving those calls after the `original_callback` didn't work as well. Instead, most of the test fails.

Reply:
That undoes the whole point of this patch though.

Reply:
Ok. I'll look deep into the failing tests.